### DISCUSSION TITLE
Issue with Oauth overwriting users

### DISCUSSION THREAD

**Discussion by ARCHES-Academy** [statamic/cms#4367] [Troubleshooting] [answered]
I have Oauth implemented on our website, but I'm having some issues.

In order for everything on the website to work correctly, I need to add the users in bulk before they actually log in. (Student accounts need to exist in order to assign work, classes, and grades to them.)

However, when the user logs in (with Oauth), instead of finding the existing user, it creates a new user with the same email, which overwrites the old one when I initiate a git push/pull. All the data that was saved with the user I created to begin with (name, bio, etc.) is deleted, in favor of the null values pulled from Oauth.

Is there a way to allow Oauth to find existing users but not create new ones?

**Comment by duncanmcclean** [comment, marked as answer]
Duplicate of #3269

**Reply by ARCHES-Academy** [reply]
Thanks. I did some searching, but for some reason I didn't see that one.

**Reply by duncanmcclean** [reply]
No worries!
